book/src/dev/rfcs/0005-state-updates.md

```diff
@@ -822,6 +821,8 @@ So they should not be used for consensus-critical checks.
   regardless of where they come from. The exception is `sprout_anchors` which also maps
   the anchor to the matching note commitment tree. This is required to support interstitial
   treestates, which are unique to Sprout.
+  **TODO:** store the `Root` hash in `sprout_note_commitment_tree`, and use it to look up the
+  note commitment tree. This de-duplicates tree state data.
```

Comment:
Note (to self): We currently store at most one tree in `sprout_note_commitment_tree`, so this optimization would save at most one tree.

Reply:
Ah right yeah that's why I put this optimisation at the bottom of that ticket.
```suggestion
  **TODO:** store the `Root` hash in `sprout_note_commitment_tree`, and use it to look up the
  note commitment tree. This de-duplicates tree state data. But we currently only store one sprout tree by height.
```